db/upgrade.php

```diff
@@ -558,5 +557,23 @@ function xmldb_local_recompletion_upgrade($oldversion) {
         upgrade_plugin_savepoint(true, 2021091500, 'local', 'recompletion');
     }
 
+    if ($oldversion < 2021100101) {
+        // Define table local_recompletion_qr to be created.
+        $table = new xmldb_table('local_recompletion_qr');
+        $field = new xmldb_field('course', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, '0', 'userid');
+        if (!$dbman->field_exists($table, $field)) {
```

Comment:
Can you please minimise this to the individual issue you found with the default value? - precision etc shouldn't need to change.

Reply:
I've updated this to just apply the expected default. The field was never 'missing' so I've removed the surrounding conditions as well.